api/graphql/schema.graphql

```diff
@@ -256,7 +256,9 @@ type EmployeeProfile {
   careerObjectiveCSuiteRoleTitle: LocalizedCSuiteRoleTitle
     @rename(attribute: "career_objective_c_suite_role_title")
 
-  communityInterests(id: UUID @eq): [CommunityInterest!] @belongsToMany
+  communityInterests(id: UUID @eq): [CommunityInterest!]
+    @belongsToMany
+    @canResolved(ability: "view")
```

Comment:
OK, so this works!  (Yay!) But it means that there's one more thing to do that I didn't anticipate. (Boo!)  When the community talent coordinator goes to view the user's profile, it is still trying to load all the community interests but at least the guard is crashing the query before it can divulge that data.  
![image](https://github.com/user-attachments/assets/b0f99c6e-789b-4d2e-876d-31f5117a9f14)
I think the extra step needed now is to add an "authroizedToView" scope so that only the interests that will pass the policy (in this case, belong to the coordinator's communities) are returned.  A very similar thing happened in Matt's PR https://github.com/GCTC-NTGC/gc-digital-talent/pull/13118 if you want to reference.  Let me know if you want to chat about it.

Reply:
Added with [Add authorizedToView scope](https://github.com/GCTC-NTGC/gc-digital-talent/pull/13130/commits/dfa2efb4b504b7b2e5a5a53396ad61195ad77407).

Reply:
OK, the talent coordinator can see the limited list now.  But the owner can't!  This scope is missing logic for the owner to be able to see all of their own interests, like the policy has.

Reply:
This _should_ be fixed with [Add own users interests to authorised scope](https://github.com/GCTC-NTGC/gc-digital-talent/pull/13130/commits/d82bac15140a92979235819f8c452efabf8fa153).

Reply:
Added [Improve fallback in authorizedToView scope](https://github.com/GCTC-NTGC/gc-digital-talent/pull/13130/commits/300a90a6e3be20189b26456bf17c7cb968568a48) to improve this.